lib/Service/ProvidersAI/TaskProcessingService.php

```diff
@@ -81,11 +84,18 @@ public function registerTaskProcessingProvider(
 			$taskProcessingProvider = null;
 		}
 		try {
+			if ($customTaskType !== null && $taskType !== $customTaskType['id']) {
+				throw new Exception('Task type and custom task type must be the same if custom task type is provided');
+			} elseif ($customTaskType === null && $provider['task_type'] === null) {
+				throw new Exception('Task type must be provided if custom task type is not provided');
+			}
+
```

Comment:
I think it would be good to have more validation here to give early feedback to the exapp that the registration failed. Otherwise if the passed array is wrong, it will return ok, but when we want to use the provider it will fail without the exapp getting wind of it.